plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/e4/DBeaverStackRenderer.java

```diff
@@ -49,21 +54,55 @@
 import org.jkiss.dbeaver.ui.ActionUtils;
 import org.jkiss.dbeaver.ui.ShellUtils;
 import org.jkiss.dbeaver.ui.UIUtils;
+import org.jkiss.dbeaver.ui.controls.HolidayDecorations;
 import org.jkiss.dbeaver.ui.editors.EditorUtils;
 import org.jkiss.dbeaver.ui.editors.IDatabaseEditorInput;
 import org.jkiss.dbeaver.ui.editors.sql.SQLEditor;
 import org.jkiss.dbeaver.ui.editors.sql.SQLEditorCommands;
 import org.jkiss.dbeaver.ui.editors.sql.SQLEditorUtils;
 import org.jkiss.dbeaver.ui.editors.sql.handlers.SQLEditorHandlerRenameFile;
 import org.jkiss.dbeaver.ui.editors.sql.internal.SQLEditorMessages;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
 
+import javax.inject.Inject;
 import java.io.File;
+import java.lang.reflect.Field;
 
 
 public class DBeaverStackRenderer extends StackRenderer {
 
     private static final Log log = Log.getLog(DBeaverStackRenderer.class);
 
+    @Inject
+    @Optional
+    void subscribePerspectiveSwitched(@NotNull IEventBroker broker) {
+        final EventHandler handler = new EventHandler() {
+            @Override
+            public void handleEvent(Event event) {
+                final Object element = event.getProperty(UIEvents.EventTags.ELEMENT);
```

Comment:
Let's protect it with try/catch throwable.
I am afraid that old Eclipse version or some other thing may cause an unexpected error and may break entire app.